### PR TITLE
Euclid filters table UI support

### PR DESCRIPTION
**Pull request by dsavchenko**

### CODE REVIEW

**Comment by dsavchenko** [comment]
@motame I tried to get rid of hardcoded parameter name for the Euclid table. 
The encoding part works as expected. The validation however doesn't work well, it validates only the first line of the table. Could you help me with this?

Attached is the Euclid tab as generated with https://github.com/oda-hub/frontend-tab-generator/pull/23 
for you to test (I deleted your original example from branch to clean up the PR).

[mmoda_euclid.zip](https://github.com/oda-hub/mmoda-frontend-module/files/14167736/mmoda_euclid.zip)

File: js/mmoda.instrument.js [hunk 1]
```diff
@@ -1124,22 +1112,31 @@ function panel_title(srcname, param) {
         // true);
 
         // Collect object name 
-        var allFormData = $("form#mmoda-name-resolve").serializeArray().map(function(item, index) {
+        var allFormData = $("form#mmoda-name-resolve").serializeArray().map(function(item) {
```
Comment (dsavchenko) [inline js/mmoda.instrument.js:1115]: Hm, these removals of `index` are not my changes, they somehow appeared as a result of merging. I'm not sure if this is something critical, it's not used inside...

Comment (burnout87) [reply js/mmoda.instrument.js:1115]: I confirm they are from the latest changes @motame introduced, I also had to merge those changes today

**Comment by motame** [comment]
OK, I am on it and let you know soon.

**Comment by motame** [comment]
@dsavchenko, the issue is fixed and pushed in [euclid-filter-select-ui](https://github.com/oda-hub/mmoda-frontend-module/tree/euclid-filter-select-ui)

File: js/mmoda.instrument.js [hunk 2]
```diff
@@ -540,22 +540,28 @@ function panel_title(srcname, param) {
...
         }
       }
+<<<<<<< HEAD
+    };
+    //$('.instrument-panel form').bootstrapValidator('destroy');
+=======
     }
 
     $('.instrument-panel form').bootstrapValidator('destroy');
+>>>>>>> 867ac8a017c8b88e13550ea8b419ff6105606512
```
[the remark this answers is not in the document]
Comment (dsavchenko) [reply js/mmoda.instrument.js]: Weird, it's not actually in the branch. Seems this is some bug in github PR diff preview

**Comment by dsavchenko** [comment]
> It is almost a copy of the zip file you provided [mmoda_euclid.zip](https://github.com/oda-hub/mmoda-frontend-module/files/14167736/mmoda_euclid.zip).

Thanks, I will check differences and get back to you if unclear

File: js/mmoda.instrument.js [hunk 3]
```diff
@@ -1124,22 +1095,30 @@ function panel_title(srcname, param) {
...
             item.value = item.value.replace(' ', 'T')
           }
           return (item);
         });
         let active_panel_instrument = $('input[name=instrument]', ".instrument-panel.active").val();
+        var instrument_form_serializeJSON = $($(this)[0]).serializeJSON();
+        //var instrument_form_serializeArray = $($(this)[0]).serializeArray();
+        var instrument_form_serializeArray = [];
+        $.each(instrument_form_serializeJSON, function(param, value) {
+          if (param == 'filters') instrument_form_serializeArray.push({ 'name': param, 'value': JSON.stringify(value) });
```
Comment (dsavchenko) [inline js/mmoda.instrument.js:1114]: Here, the hardcoded parameter name came back. I did changes in this commit https://github.com/oda-hub/mmoda-frontend-module/pull/228/commits/8e7a1238d667002f947cbeea3c2f7cb8b9243a95

**Review by dsavchenko — COMMENTED** [review]
This PR went out-of-sync with the master branch (despite master is merged here) in sense that there are many differences that are unrelated to the multivalued field. 
E.g. 
- mmoda.common.js should be the same as in master, I believe.
- files are deleted from mmoda_isgri/help_book
- mmoda.instrument.js in this branch is probably based on some ancient state without new fixes merged in master

Also, some of my changes disappeared with your latest commits. See e.g. comment above on the hardcoded parameter name

Could you please reconcile this branch so that only related changes are there? @motame

File: js/mmoda.instrument.js [hunk 4]
```diff
@@ -302,18 +152,97 @@ function panel_title(srcname, param) {
...
+            });
+          }
+
+          current_ajax_call_params.currentFormData = cloneFormData(current_ajax_call_params.initialFormData);
+          current_ajax_call_params.currentFormData.append('query_status', data.query_status);
+          if (!current_ajax_call_params.currentFormData.has('job_id')) {
+            current_ajax_call_params.currentFormData.append('job_id', job_id);
+            current_ajax_call_params.currentFormData.append('session_id', session_id);
+          }
+          requestTimer = setTimeout(AJAX_call, 5000);
```
Comment (burnout87) [inline js/mmoda.instrument.js:196]: also this whole piece of code is from the an version of the file, is not in the master
same issue is noticed elsewhere

it seems like an old version of the file has been pushed

**Review by burnout87 — COMMENTED** [review]
probably an old commit has been pushed

**Comment by motame** [comment]
I have not seen that you merged master in euclid-filter-select-ui.

**Comment by motame** [comment]
The above issues have been fixed.
I reviewed all the changes and it seems OK now.
Would you please have a look again ?

File: js/mmoda.common.js [hunk 5]
```diff
@@ -642,6 +642,18 @@ function get_waitingDialog($modal_dialog) {
...
+    //      $("span.oda-icon-label").hide("slow")
+    //    }, 5000);
+
+    //    $(".main-toolbar").focusin(function() {
+    //      $("span.oda-icon-label").show("slow")
+    //    });
+    //    $(".main-toolbar").focusOut(function() {
+    //      $("span.oda-icon-label").hide("slow")
+    //    });
+
```
Comment (burnout87) [inline js/mmoda.common.js:656]: I have never seen this code before, what was it used for?
Can we just remove it ?

**Review by burnout87 — CHANGES_REQUESTED** [review]
some code to remove

File: js/mmoda.common.js (hunk 5 above)
Comment (motame) [reply js/mmoda.common.js]: Done !

**Review by burnout87 — APPROVED** [review]

**Comment by dsavchenko** [comment]
I opened this PR, so I can't approve. But I confirm it works. @motame I'm not sure if your main repo is here or in gitlab. Can you merge?

**Comment by motame** [comment]
> I opened this PR, so I can't approve. But I confirm it works. @motame I'm not sure if your main repo is here or in gitlab. Can you merge?
I do have two remote repos : GitHub et Gitlab and that was the source of this trouble :-)
I was sure that the default remote one was GitHub but at one time it changed and that caused the troubles you mentioned here.
I'll get rid soon of the Gitlab one ...

**Pull request merged**